### PR TITLE
bpo-18374: fix wrong col_offset of some ast.BinOp instances

### PR DESCRIPTION
**Pull request by cfbolz**
Nested BinOp instances (e.g. a+b+c) had a wrong col_offset for the second BinOp (e.g. 2 instead of 0 in the example). Fix it by using the correct st node to copy the line and col_offset from in ast.c.

https://bugs.python.org/issue18374

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Our records indicate we have not received your CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by asottile — APPROVED** [review]
the output looks correct

validating this change using [astpretty](https://github.com/asottile/astpretty)

### before

```console
$ astpretty /dev/stdin <<< '1 + 2 + 3'
Module(
    body=[
        Expr(
            lineno=1,
            col_offset=0,
            end_lineno=1,
            end_col_offset=9,
            value=BinOp(
                lineno=1,
                col_offset=6,
                end_lineno=1,
                end_col_offset=9,
                left=BinOp(
                    lineno=1,
                    col_offset=0,
                    end_lineno=1,
                    end_col_offset=5,
                    left=Constant(lineno=1, col_offset=0, end_lineno=1, end_col_offset=1, value=1, kind=None),
                    op=Add(),
                    right=Constant(lineno=1, col_offset=4, end_lineno=1, end_col_offset=5, value=2, kind=None),
                ),
                op=Add(),
                right=Constant(lineno=1, col_offset=8, end_lineno=1, end_col_offset=9, value=3, kind=None),
            ),
        ),
    ],
    type_ignores=[],
)
```

roughly looks like this

```python
1 + 2 + 3
      AAA        # outer BinOp
BBBBB            # inner BinOp
```

### after

```console
$ astpretty /dev/stdin <<< '1 + 2 + 3'
Module(
    body=[
        Expr(
            lineno=1,
            col_offset=0,
            end_lineno=1,
            end_col_offset=9,
            value=BinOp(
                lineno=1,
                col_offset=0,
                end_lineno=1,
                end_col_offset=9,
                left=BinOp(
                    lineno=1,
                    col_offset=0,
                    end_lineno=1,
                    end_col_offset=5,
                    left=Constant(lineno=1, col_offset=0, end_lineno=1, end_col_offset=1, value=1, kind=None),
                    op=Add(),
                    right=Constant(lineno=1, col_offset=4, end_lineno=1, end_col_offset=5, value=2, kind=None),
                ),
                op=Add(),
                right=Constant(lineno=1, col_offset=8, end_lineno=1, end_col_offset=9, value=3, kind=None),
            ),
        ),
    ],
    type_ignores=[],
)
```

roughly looks like this

```python
1 + 2 + 3
AAAAAAAAA        # outer BinOp
BBBBB            # inner BinOp
```

**Comment by ilevkivskyi** [comment]
@cfbolz Did you sign the CLA?

**Comment by cfbolz** [comment]
@ilevkivskyi yes, signed it last Friday when I opened the pull request.

**Review by ilevkivskyi — APPROVED** [review]
Thanks! LGTM.

File: Lib/test/test_ast.py [hunk 1]
```diff
@@ -576,6 +576,36 @@ def bad_normalize(*args):
...
+        grandchild_binop = child_binop.left
+        self.assertEqual(parent_binop.col_offset, 0)
+        self.assertEqual(parent_binop.lineno, 1)
+        self.assertEqual(parent_binop.end_col_offset, 4)
+        self.assertEqual(parent_binop.end_lineno, 2)
+
+        self.assertEqual(child_binop.col_offset, 0)
+        self.assertEqual(parent_binop.lineno, 1)
+        self.assertEqual(child_binop.end_col_offset, 2)
+        self.assertEqual(parent_binop.end_lineno, 2)
```
Comment (ilevkivskyi) [inline Lib/test/test_ast.py:603]: Oh, after merging this I noticed this block of asserts checks parent for `lineno` and `end_lineno`.

File: Lib/test/test_ast.py [hunk 2]
```diff
@@ -576,6 +576,36 @@ def bad_normalize(*args):
...
+
+        self.assertEqual(grandchild_binop.col_offset, 0)
+        self.assertEqual(parent_binop.lineno, 1)
+        self.assertEqual(grandchild_binop.end_col_offset, 3)
+        self.assertEqual(parent_binop.end_lineno, 2)
```
Comment (ilevkivskyi) [inline Lib/test/test_ast.py:608]: Same in this block, it should be `granchild_binop` in two asserts in this block. @cfbolz could you please make another PR with a fix?

**Comment by cfbolz** [comment]
Thanks a lot, Ivan and Anthony!

**Comment by cfbolz** [comment]
Yep, will fix the problems and ping you.
